examples/tests/invalid/function_call_arity.test.w

```diff
@@ -0,0 +1,39 @@
+
+let func = (n: num, b: bool) => {};
+
+func(); // Expected {} positional arguments but got {}
+
+let optionalParamsFunc = (n: num, b: bool, s: str?) => {};
+
+optionalParamsFunc(1); // Expected between {} and {} positional arguments but got {}
+
+let variadicFunc = (n: num, b: bool, ...events: Array<Json>) => {};
+
+variadicFunc(); // Expected {} positional arguments + variadic args but got {}
+
+
+let variadicOptionalParamsFunc = (n: num, b: bool, s: str?, ...events: Array<Json>) => {};
+
+variadicOptionalParamsFunc(2); // Expected between {} and {} positional arguments + variadic args but got {}
+
+struct Options {
+  prefix: str;
+  priority: num;
+}
+
+let namedFunc = (n: num, b: bool, opts: Options) => {};
+
+namedFunc(); // Expected {} positional arguments or named arguments for the last parameter but got {}
+
+let namedOptionalParamsFunc = (n: num, b: bool, s: str?, opts: Options) => {};
+
+namedOptionalParamsFunc(3); // Expected between {} and {} positional arguments or named arguments for the last parameter but got {}
+
+
+// variadic and named args cannot be combined atm
+let invalidCombinationFunc = (n: num, b: bool, ...variadic: Array<str>, opts: Options) => {};
```

Comment:
just to be clear this :point_up: give an error even without trying to call it?
`invalidCombinationFunc(1,2,"hi", prefix: "x", priority: 0);`

Reply:
yes as we have checks for variadics only allowed as the last parameter

Reply:
we dont seem to have that for the named args for structs, thats why i added a call for it